sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -466,20 +453,8 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
             // aggregates have different column expressions.
             val distinctExpressions =
               functionsWithDistinct.head.aggregateFunction.children.filterNot(_.foldable)
-            val normalizedNamedDistinctExpressions = distinctExpressions.map { e =>
-              // Ideally this should be done in `NormalizeFloatingNumbers`, but we do it here
-              // because `distinctExpressions` is not extracted during logical phase.
-              NormalizeFloatingNumbers.normalize(e) match {
-                case ne: NamedExpression => ne
-                case other =>
-                  // Keep the name of the original expression.
-                  val name = e match {
-                    case ne: NamedExpression => ne.name
-                    case _ => e.toString
-                  }
-                  Alias(other, name)()
-              }
-            }
+            val normalizedNamedDistinctExpressions =
+              AggUtils.normalizeDistinctGroupingExprs(distinctExpressions)
```

Comment:
`normalizeDistinctGroupingExprs` is called only once, but I moved this logic into the `AggUtils` side so that the similar logics are located in close places.